src/wallet/output_alias.c

```diff
@@ -0,0 +1,423 @@
+  if (ret != 0) {
+    printf("[%s:%d] create alias output failed\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // check if reminder is needed
+  if (total_unspent_amount > send_amount) {
+    ret = wallet_output_basic_create(sender_addr, total_unspent_amount - send_amount, tx->essence);
+    if (ret != 0) {
+      printf("[%s:%d] create a reminder basic output failed\n", __func__, __LINE__);
+      goto end;
+    }
+  }
+
+  // create a core message
+  core_message_t* message = wallet_create_core_message(w, tx, unspent_outputs, sign_data);
+  if (!message) {
+    printf("[%s:%d] can not create a core message\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // calculate transaction payload ID
+  byte_t payload_id[CRYPTO_BLAKE2B_256_HASH_BYTES] = {0};
+  if (tx_payload_calculate_id(tx, payload_id, sizeof(payload_id)) != 0) {
+    ret = -1;
+    printf("[%s:%d] can not calculate transaction payload ID\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // create alias output ID
+  byte_t output_id[IOTA_OUTPUT_ID_BYTES] = {0};
+  memcpy(output_id, payload_id, sizeof(payload_id));
+  memset(output_id + sizeof(payload_id), 0,
+         sizeof(uint16_t));  // index is always 0 because in this function alias output is added into a transaction
+                             // essence before basic output
+
+  // create alias address from alias output ID
+  if (alias_address_from_output(output_id, sizeof(output_id), alias_addr) != 0) {
+    ret = -1;
+    printf("[%s:%d] can not create alias address from output Id!\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // send a message to a network
+  ret = wallet_send_message(w, message, msg_res);
+
+  // clean up memory
+  core_message_free(message);
+
+end:
+  signing_free(sign_data);
+  utxo_outputs_free(unspent_outputs);
+  return ret;
+}
+
+int wallet_alias_state_transition_send(iota_wallet_t* w, byte_t alias_id[], address_t* state_ctrl_addr,
+                                       ed25519_keypair_t* state_ctrl_keypair, address_t* govern_addr,
+                                       res_send_message_t* msg_res) {
+  if (w == NULL || alias_id == NULL || state_ctrl_addr == NULL || state_ctrl_keypair == NULL || govern_addr == NULL ||
+      msg_res == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  int ret = 0;
+  utxo_outputs_list_t* unspent_outputs = NULL;
+  res_output_t* output_res = NULL;
+  signing_data_list_t* sign_data = signing_new();
+
+  // create a tx
+  transaction_payload_t* tx = tx_payload_new(w->network_id);
+  if (!tx) {
+    printf("[%s:%d] create tx payload failed\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // get unspent alias output
+  output_res = wallet_get_unspent_alias_output(w, alias_id);
+  if (!output_res) {
+    printf("[%s:%d] alias address does not have any unspent alias outputs\n", __func__, __LINE__);
+    ret = -1;
```

Comment:
the same as `wallet_alias_create_send`, the `tx` memory will be owned by the message, but before message creation, the `tx` has not been freed here, also following exceptions.